darkdetect/_mac_detect.py

```diff
@@ -69,6 +82,37 @@ def isDark():
 def isLight():
     return theme() == 'Light'
 
-#def listener(callback: typing.Callable[[str], None]) -> None:
-def listener(callback):
-    raise NotImplementedError()
+
+def _listen_child():
+    """
+    Run by a child process, install an observer and print theme on change
+    """
+    class Observer(NSObject):
+        def callback_(self, _):
+            try:
+                print(theme(), flush=True)
+            except IOError:
+                os._exit(1)
+    observer = Observer.new()  # Keep a reference of the observer to keep it alive
+    NSDistributedNotificationCenter.defaultCenter().addObserver_selector_name_object_(
+        observer,  # Observer must be kept alive by a different reference
+        "callback:",
+        "AppleInterfaceThemeChangedNotification",
+        None
+    )
+    AppHelper.runConsoleEventLoop()
+
+
+def listener(callback: Callable[[str], None]) -> None:
+    if not _can_listen:
+        raise NotImplementedError()
+    sig = "import signal as s; s.signal(s.SIGINT, s.SIG_IGN)"
```

Comment:
Is this truly necessary? With my implementation, I was able to interrupt the subprocess even without it:
```python
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alberto/darkdetect_zwimer/darkdetect/_mac_detect.py", line 132, in listener
    for line in p.stdout:
KeyboardInterrupt
```

Perhaps the key is to use `AppHelper.runConsoleEventLoop(installInterrupt=True)`.

Reply:
The reason I did this is to fix the problem where: the main program catches `KeyboardInterrupt` and the listener shouldn't die as a result. Depending if you are in a terminal or not, this signal may end up being sent to the listener process, which we probably do not want.

It also has a bonus of ensuring no log spam when the listener dies for a KeyboardInterrupt, but that is a side effect, not the reason.

Reply:
I honestly do not see any difference with my implementation, but this should not be harmful. Another possibility is to add the signal ignore code in the beginning of `_listen_child`.

Reply:
For me the PR would be complete as this is not actually a requirement. However, because the signal addition does not seem to do anything, I would remove it from the code.

Reply:
This was caused by a bug; I've fixed it.